Hearthstone/app/src/main/java/com/example/hearthstone/ui/searchpage/SearchPageFragment.kt

```diff
@@ -0,0 +1,113 @@
+package com.example.hearthstone.ui.searchpage
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.databinding.DataBindingUtil
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentActivity
+import androidx.navigation.findNavController
+import androidx.navigation.fragment.navArgs
+import com.example.hearthstone.MainActivity
+import com.example.hearthstone.R
+import com.example.hearthstone.adapter.HearthStoneAdapterSP
+import com.example.hearthstone.data.network.repo.HSRepo
+import com.example.hearthstone.databinding.FragmentSearchPageBinding
+import com.example.hearthstone.dialogue.ErrorCardName
+
+
+class SearchPageFragment : Fragment() {
+
+    private lateinit var binding: FragmentSearchPageBinding
+    private lateinit var viewModel: SearchPageViewModel
+    private val args: SearchPageFragmentArgs by navArgs()
+
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        // Inflate the layout for this fragment
+        binding = DataBindingUtil.inflate(
+            inflater, R.layout.fragment_search_page,
+            container, false
+        )
+        return binding.root
+    }
+
+    @SuppressLint("SetTextI18n")
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        goHome()
+        buttonSearch()
+        viewModel =
```

Comment:
are you going to be doing DI soon?

Reply:
Yes, the DI is in the next PR in which I only included the DP